### DISCUSSION TITLE
How to get bounds

### DISCUSSION THREAD

**Discussion by BarbourSmith** [sgenoud/replicad#167] [Q&A] [answered]
I see that bounding box seems to be implemented, but I'm not quite sure how to use it in practice. 

Edit: Here is the docs that I'm looking at: https://replicad.xyz/docs/api/classes/BoundingBox

Lets say that I have something like the vase example and I want to get the height, how would I do something like that?

```
const { draw } = replicad;

const defaultParams = {
  height: 100,
  baseWidth: 20,
  wallThickness: 5,
  lowerCircleRadius: 1.5,
  lowerCirclePosition: 0.25,
  higherCircleRadius: 0.75,
  higherCirclePosition: 0.75,
  topRadius: 0.9,
  topFillet: true,
  bottomHeavy: true,
};

const main = (
  r,
  {
    height,
    baseWidth,
    wallThickness,
    lowerCirclePosition,
    lowerCircleRadius,
    higherCircleRadius,
    higherCirclePosition,
    topRadius,
    topFillet,
    bottomHeavy,
  }
) => {
  const splinesConfig = [
    { position: lowerCirclePosition, radius: lowerCircleRadius },
    {
      position: higherCirclePosition,
      radius: higherCircleRadius,
      startFactor: bottomHeavy ? 3 : 1,
    },
    { position: 1, radius: topRadius, startFactor: bottomHeavy ? 3 : 1 },
  ];

  const sketchVaseProfile = draw().hLine(baseWidth);

  splinesConfig.forEach(({ position, radius, startFactor, endFactor }) => {
    sketchVaseProfile.smoothSplineTo([baseWidth * radius, height * position], {
      endTangent: [0, 1],
      startFactor,
      endFactor,
    });
  });

  let vase = sketchVaseProfile
    .lineTo([0, height])
    .close()
    .sketchOnPlane("XZ")
    .revolve();

  if (wallThickness) {
    vase = vase.shell(wallThickness, (f) => f.containsPoint([0, 0, height]));
  }

  if (topFillet) {
    vase = vase.fillet(wallThickness / 3, (e) => e.inPlane("XY", height));
  }
  console.log(vase.height());
  return vase;
};
```

**Comment by BarbourSmith** [comment, marked as answer]
The answer is geometry[0].boundingBox.bounds[1][2] 😁 

For anyone else who stumbles across this in a google search 😛

**Reply by sgenoud** [reply]
There is a nicer way to do it!

You can for instance do:

```js
vase.boundingBox.height
```

Have a look [here](https://studio.replicad.xyz/workbench#code=UEsDBAoAAAAIAJs2O1m90STHNQIAALgFAAAHAAAAY29kZS5qc4VUW2vbMBR%252Bz6845MkeRnNaxlhKN1hZ2cNgYQu7hTyotmyLKlKQlGQj%252BL%252F3WLJlu07ZSxKd23eRTjIljYUz5JqeoIZb0GwveEbzm9ksc7mcFfQg7IpqujNYcJ4BVIyXlV3CIk0TPD5Qw37y3FZLuHKBExViXfHsUTJjlvCmiQl1YvqO60ywbzTnB4wvyPPMShluuZJLSMmVS1aI9LwvJW8nuWGnz1q17xvetZF7LgRD5lYfmKOurFW7z4we%252F3XBOkjfUS5RcYR1uilupHfiE%252Fc7KPfHkW4fuiBukvA0fXiqdxofDwo6w9GLbBn2AptAPYvh9r1T4jUavG5ke6dkwUsUu3FdZ9gHPy8pAN06O1EBtcf1XsFgzssKIIx7ST2AsVTbe5pZpZdDUfABrgGfkq%252FrwAewi55s79R%252Fx%252FlJW3wKwahHZrPqB174SquCC4ZmNVsTxaT6ghZG4S3ErmtkLCmU%252FkSzKop6ah2vEZkEmMz9T6jDVcEUnZidUrb67lDWKtoEeHgVBvunioEOc5uEi0GcNZUlk7gNmzSBxfaC1UlfPAzUKNF%252FNuvLLBwRHP2YsHTVpKOYdoy2sU9kQhkWtQff%252FFWuBEU357%252F%252BzNuEZkcljljn4HgB0WjP4lZSy6H5IqZiQozLEogKZ2hB8EItrrZZKS6tozVg5qV1SGGbLqEULjOGgddwjVDMQTHCg5zf8w6jgwBoH5fC6xSqjNzQB3WQOZflR%252FWXtPVNuWb2oKXDvcG%252FqCdQSwECFAAKAAAACACbNjtZvdEkxzUCAAC4BQAABwAAAAAAAAAAAAAAAAAAAAAAY29kZS5qc1BLBQYAAAAAAQABADUAAABaAgAAAAA%253D)

**Reply by BarbourSmith** [reply]
Super elegant 😀 You are already two steps ahead.
